src/coreclr/jit/llvm.cpp

```diff
@@ -1654,14 +1630,190 @@ void startImportingNode(llvm::IRBuilder<>& builder)
     }
 }
 
+
+Llvm::Llvm(Compiler* pCompiler)
+{
+    _compiler = pCompiler;
+    _info = pCompiler->info;
+
+    _function = nullptr;
+    _requiresShadowStackAddSubsitution = false;
+    _sigInfoIsValid = false;
+}
+
+void Llvm::llvmShutdown()
+{
+    if (_diBuilder != nullptr)
+    {
+        emitDebugMetadata(_llvmContext);
+    }
+#if DEBUG
+    if (_outputFileName == nullptr) return; // nothing generated
+    std::error_code ec;
+    char* txtFileName = (char*)malloc(strlen(_outputFileName) + 2); // .txt is longer than .bc
+    strcpy(txtFileName, _outputFileName);
+    strcpy(txtFileName + strlen(_outputFileName) - 2, "txt");
+    llvm::raw_fd_ostream textOutputStream(txtFileName, ec);
+    _module->print(textOutputStream, (llvm::AssemblyAnnotationWriter*)NULL);
+    free(txtFileName);
+#endif //DEBUG
+    llvm::raw_fd_ostream OS(_outputFileName, ec);
+    llvm::WriteBitcodeToFile(*_module, OS);
+    delete _module;
+    //    Module.Verify(LLVMVerifierFailureAction.LLVMAbortProcessAction);
+}
+
+GenTree* createAddNodeForShadowStackLocal(LclVarDsc* varDsc, unsigned shadowStackLclNum, GenTreeIntCon*& offset, GenTreeLclVar*& shadowStackVar)
+{
+    // TODO-LLVM: if the offset == 0, just GT_STOREIND at the shadowStack
+    offset = _compiler->gtNewOneConNode(var_types::TYP_INT)->AsIntCon();
+    offset->SetIconValue(varDsc->GetStackOffset());
+    shadowStackVar = _compiler->gtNewLclvNode(shadowStackLclNum, var_types::TYP_REF);
```

Comment:
```suggestion
    shadowStackVar = _compiler->gtNewLclvNode(shadowStackLclNum, TYP_I_IMPL);
```
The shadow stack base itself is not an object reference...